# Error estimate for T-boosted Langevin simulations

* Import external libraries:

In [1]:
import numpy as np

**Example trypsin**

In [2]:
# number of observed transitions
no_e = np.array([92, 148, 306, 712, 2153, 5141, 10534, 18818, 30163, 44864,
                 62538, 82528, 105127])

# used temperatures in [K]
T = np.array([380, 400, 420, 450, 500, 550, 600, 650, 700, 750, 800, 850, 900])
TSqr = T**2

# Calculate ...
delta = np.sum(no_e) * np.sum(no_e/TSqr) - np.sum(no_e/T)**2

* error estimate

In [3]:
# Use error propagation to derive error at 300 K.
T_e = 300  # [K]
TSqr_e = T_e**2

error = np.sqrt((np.sum(no_e)/TSqr_e + np.sum(no_e/TSqr)) / delta)
error_slope = np.sqrt((np.sum(no_e)/TSqr_e) / delta)

# Complete error in %
print('error: {:.2%}'.format(error))

# Error only from slope in %, we see that it dominates the error
print('error slope: {:.2%}'.format(error_slope))

error: 3.34%
error slope: 3.12%


This means that we observe an error of 3.3 % for Trypsin

**Example Hsp90**

In [4]:
# number of observed transitions
no_e = np.array([77, 204, 422, 800, 1376, 2147, 3265, 4796, 6861, 9229,
                 12233, 15753, 19832, 24496])

# used temperatures in [K]
T = np.arange(700, 1400, 50)
TSqr = T**2

# Calculate ...
delta = np.sum(no_e) * np.sum(no_e/TSqr) - np.sum(no_e/T)**2

* error estimate

In [5]:
# Use error propagation to derive error at 300 K.
T_e = 300  # [K]
TSqr_e = T_e**2

error = np.sqrt((np.sum(no_e)/TSqr_e + np.sum(no_e/TSqr)) / delta)
error_slope = np.sqrt((np.sum(no_e)/TSqr_e) / delta)

# Complete error in %
print('error: {:.2%}'.format(error))

# Error only from slope in %, we see that it dominates the error
print('error slope: {:.2%}'.format(error_slope))

error: 11.03%
error slope: 10.70%


This means that we have 11% uncertaintiy for Hsp90.